File: src/main/java/social/alone/server/event/EventValidator.java
package social.alone.server.event;


import org.springframework.stereotype.Component;
import org.springframework.validation.Errors;
import social.alone.server.event.dto.EventDto;

@Component
public class EventValidator {

  public void validate(EventDto eventDto, Errors errors) {

    if (
            eventDto.getStartedAt() != null &&
            eventDto.getStartedAt().isAfter(eventDto.getEndedAt())) {
      errors.reject(
              "Wrong event.startedAt with endedAt",
              "이벤트 시작 시간은 종료시간보다 뒤 일 수 없습니다."
      );
    }
  }
}

File: src/main/kotlin/social/alone/server/event/dto/EventValidator.kt
package social.alone.server.event.dto


import org.springframework.stereotype.Component
import org.springframework.validation.Errors

@Component
class EventValidator {

    fun validate(eventDto: EventDto, errors: Errors) {

        if (eventDto.startedAt != null && eventDto.startedAt!!.isAfter(eventDto.endedAt!!)) {
            errors.reject(
                    "Wrong event.startedAt with endedAt",
                    "이벤트 시작 시간은 종료시간보다 뒤 일 수 없습니다."
            )
        }
    }
}
